pkg/generate/app/cmd/newapp.go

```diff
@@ -586,7 +586,18 @@ func (c *AppConfig) validate() (app.Environment, app.Environment, error) {
 		return nil
 	})
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+	buildEnv, err := app.ParseAndCombineEnvironment(c.BuildEnvironment, c.BuildEnvironmentFiles, c.In, func(key, file string) error {
```

Comment:
I don't see where you're including the value of the "--env" parameter when running "oc new-build".